File: Order-Processing-Frontend/src/app/models/Book.ts
import { Author } from "./Author";
import { Publisher } from "./Publisher";

export class Book {
    isbn: number;
    title: string;
    publisher_id: number;
    publication_year: number;
    price: number;
    category: string;
    quantity: number;
    threshold: number;
    publisher: Publisher;
    authors: Author[];

    constructor() {
        this.isbn = -1;
        this.title = '';
        this.publisher_id = -1;
        this.publication_year = -1;
        this.price = -1;
        this.category = '';
        this.quantity = -1;
        this.threshold = -1;
        this.publisher = new Publisher();
        this.authors = [];
    }
}

File: Order-Processing-Backend/src/main/java/csed/database/orderprocessingbackend/model/Book.java
package csed.database.orderprocessingbackend.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

@Getter
@Setter
@AllArgsConstructor
public class Book implements Serializable {
    private Long ISBN;
    private String title;
    private Long publisher_id;
    private Integer publication_year;
    private Integer price;
    private String category;
    private Integer quantity;
    private Integer threshold;

    @Override
    public String toString() {
        return ISBN +
                ", '" + title + '\'' +
                ", " + publisher_id +
                ", " + publication_year +
                ", " + price +
                ", '" + category + '\'' +
                ", " + quantity +
                ", " + threshold;
    }
}
